src/main/java/educonnect/logic/parser/FindCommandParser.java

```diff
@@ -48,18 +50,35 @@ public FindCommand parse(String args) throws ParseException {
         }
 
         Set<Predicate<Student>> predicates = new HashSet<>();
-        argMultimap.getValue(PREFIX_NAME).ifPresent(keywordName ->
-                predicates.add(new NameContainsKeywordsPredicate(keywordName))
-        );
-        argMultimap.getValue(PREFIX_STUDENT_ID).ifPresent(keywordId ->
-                predicates.add(new IdContainsKeywordsPredicate(keywordId))
-        );
-        argMultimap.getValue(PREFIX_EMAIL).ifPresent(keywordEmail ->
-                predicates.add(new EmailContainsKeywordsPredicate(keywordEmail))
-        );
-        argMultimap.getValue(PREFIX_TELEGRAM_HANDLE).ifPresent(keywordTeleHandle ->
-                predicates.add(new TelegramContainsKeywordsPredicate(keywordTeleHandle))
-        );
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String name = argMultimap.getValue(PREFIX_NAME).get().trim();
+            if (name.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_EMPTY_PREFIX_ARGUMENT, "NAME", "NAME"));
+            }
+            predicates.add(new NameContainsKeywordsPredicate(name));
+        }
+        if (argMultimap.getValue(PREFIX_STUDENT_ID).isPresent()) {
+            String studentId = argMultimap.getValue(PREFIX_STUDENT_ID).get().trim();
+            if (studentId.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_EMPTY_PREFIX_ARGUMENT, "STUDENT_ID", "STUDENT_ID"));
+            }
+            predicates.add(new IdContainsKeywordsPredicate(studentId));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            String email = argMultimap.getValue(PREFIX_EMAIL).get().trim();
+            if (email.isEmpty()) {
+                throw new ParseException(String.format(MESSAGE_EMPTY_PREFIX_ARGUMENT, "EMAIL", "EMAIL"));
+            }
+            predicates.add(new EmailContainsKeywordsPredicate(email));
+        }
+        if (argMultimap.getValue(PREFIX_TELEGRAM_HANDLE).isPresent()) {
+            String telegramHandle = argMultimap.getValue(PREFIX_TELEGRAM_HANDLE).get().trim();
+            if (telegramHandle.isEmpty()) {
+                throw new ParseException(
+                        String.format(MESSAGE_EMPTY_PREFIX_ARGUMENT, "TELEGRAM_HANDLE", "TELEGRAM_HANDLE"));
+            }
+            predicates.add(new TelegramContainsKeywordsPredicate(telegramHandle));
+        }
```

Comment:
- could consider adding parameters as static variables to avoid using magic literals
- possible implementation of SLAP here to improve code quality

Reply:
Thanks for the input! I abstract out the error messages to the message class.